R/Reporter.R

```diff
@@ -17,13 +19,57 @@ Reporter <- R6::R6Class( # nolint: object_name_linter.
     #' reporter <- Reporter$new()
     #'
     initialize = function() {
```

Comment:
Should we add a parameter on constructor that allows to define `template`s?

```suggestion
    initialize = function(template = NULL) {
      checkmate::assert_function(template, null.ok = TRUE)
      self$set_template(template)
```